src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentRefactoringService.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    public interface IDocumentRefactoringService : IWorkspaceService
+    {
+        /// <summary>
+        /// Attempts to sync a named type name and namespace based on file path. <paramref name="previousFilePath" /> is
+        /// used to determine how the policy might be applied, assuming that matching types with previous
+        /// indicates a user wants the type to match with the new location or name.
+        /// May fail if:
```

Comment:
What is returned if the operation is canceled?

Reply:
I assume this will throw an `OperationCanceledException`?

Reply:
Yes